src/test/java/com/company/elixr/springbootapplicationwithdocker/FileOperationServiceTest.java

```diff
@@ -0,0 +1,164 @@
+import com.company.elixr.springbootapplicationwithdocker.responses.SuccessResponse;
+import com.company.elixr.springbootapplicationwithdocker.responses.SuccessResponseForGetById;
+import com.company.elixr.springbootapplicationwithdocker.service.FileOperationServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mock.web.MockMultipartFile;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@RunWith(MockitoJUnitRunner.class)
+@WebMvcTest(FileOperationServiceImpl.class)
+public class FileOperationServiceTest {
+
+    List<FileInfo> fileInfoList = new ArrayList<>();
+    @MockBean
+    private FileOperationRepository fileOperationRepository;
+    @Autowired
+    private FileOperationServiceImpl fileOperationService;
+    private final FileInfo fileInfo = new FileInfo();
+
+    @BeforeEach
+    public void init() {
+        MockitoAnnotations.openMocks(this);
+        fileInfo.setId(UUID.randomUUID());
+        fileInfo.setFileName("testFile.txt");
+        fileInfo.setUserName("Nathan122");
+        fileInfo.setTimeOfUpload(LocalDateTime.now());
+
+        fileInfoList.add(fileInfo);
+    }
+
+    @Test
+    void test_SaveFile() {
+
+        MockMultipartFile sampleFile = new MockMultipartFile("file", "testFile.txt",
+                "text/plain",
+                "This is the file content".getBytes());
+        Mockito.when(fileOperationRepository.save(Mockito.any(FileInfo.class))).thenReturn(fileInfo);
+        ResponseEntity<SuccessResponse> created = fileOperationService.saveFile(sampleFile, "Natasha11");
+        Assertions.assertNotNull(created);
+        Assertions.assertEquals(HttpStatus.OK, created.getStatusCode());
+        Assertions.assertEquals(Constants.STATUS, created.getBody().getStatus());
+        Assertions.assertEquals(fileInfo.getId().getClass().getName(), created.getBody().getId().getClass().getName());
+    }
+
+    @Test
+    void test_SaveFile_InvalidFileType() {
+
+        MockMultipartFile sampleFile = new MockMultipartFile("file", "testFile.json",
+                "application/json",
+                "This is the file content".getBytes());
+        Mockito.when(fileOperationRepository.save(Mockito.any(FileInfo.class))).thenReturn(fileInfo);
+        try
+        {
+            fileOperationService.saveFile(sampleFile, "Natasha11");
```

Comment:
With the current code, the test will pass if no exception is thrown either. You need to explicitly fail the test in case no exception is thrown.

Reply:
Made sufficient changes to fail the test explicitly.